## Set up DMP params

# Parameters of the linear part of the DMP (spring-damper)
a_z: 15.0
# b_z: 5.0

DMP_TYPE: "DMP" # "DMP", "DMP-bio", "DMP-plus", "DMP-Shannon"

N_kernels: [80; 80; 80; 80; 80; 80] # number of kernels used in the DMP

kernelStdScaling: 1.0 # scaling factor for the kernels std

trainMethod: "LWR" # "LWR", "LS", "RLS" , "RLWR"

CAN_CLOCK_TYPE: "lin"

SHAPE_ATTR_GATTING_TYPE: "sigmoid" # "lin", "exp", "spring-damper", "sigmoid", "constant"
SHAPE_ATTR_GATTING_u0: 1.0 # starting value of the shape attractor gating
SHAPE_ATTR_GATTING_u_end: 0.99 # ending value of the shape attractor gating

GOAL_ATTR_GATTING_TYPE: "lin" # "lin", "exp", "spring-damper", "sigmoid", "constant"
GOAL_ATTR_GATTING_u0: 1.0 # starting value of the goal attractor gating
GOAL_ATTR_GATTING_u_end: 1.0 # ending value of the goal attractor gating

USE_GOAL_FILT: false
a_g: 20.0
USE_PHASE_STOP: true
a_px: 50.0
a_py: 40.0

# Parameters for DMP-plus
k_trunc_kernel: 3 # number of stds beyond which the kernel is truncated

# Parameters for DMP-Shannon
Wmin: 0.99
Freq_min: 0.5
Freq_max: 20.0
P1_min: 1000000000.1

## Robot controller params
Md_p: 1.0 # translational inertia
Kd_p: 0.0 # translational stiffness
Dd_p: 10.0  # translational damping
Md_o: 1.0 # rotational inertia
Kd_o: 0.0 # rotational stiffness
Dd_o: 5.5  # rotational damping

Fee_dead_zone: [0.1; 0.1; 0.12; 0.12; 0.12; 0.12]
F_norm_retrain_thres: 5

## Simulation params
pos_tol_stop: 0.01 # position error tolerance to stop the simulation
orient_tol_stop: 0.005 # orientation error tolerance to stop the simulation
tau_sim_scale: 1.0 # scaling factor for the time of the DMP simulation
goal_scale: 1.0 # scaling factor for the goal in the DMP simulation

#:===  Path to input-output data :===
binary: true
data_input_path: "/home/slifer/Desktop/slifer/as64_ws/matlab/test/DMP_test/DMP/data/"
data_output_path: "/home/slifer/Desktop/slifer/as64_ws/matlab/test/DMP_test/DMP/data/"

in_data_filename: "data.bin"
out_data_filename: "data_out"
demo_data_filename: "ur10_demo_data.bin"

in_CartPos_data_filename: "CartPos_data.bin"
out_CartPos_data_filename: "data_out"

in_orient_data_filename: "orient_data.bin"
out_orient_data_filename: "data_out"
